### PR TITLE
add configurable max blob size

### PR DESCRIPTION
**Pull request by bxue-l2**
## Why are these changes needed?

This PR makes max blob size a configurable parameters, validate if the blob is power of 2.

## Checks

- [ ] I've made sure the lint is passing in this PR.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, in that case, please comment that they are not relevant.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Integration tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: disperser/cmd/apiserver/main.go [hunk 1]
```diff
@@ -114,6 +115,14 @@ func RunDisperserServer(ctx *cli.Context) error {
 		ratelimiter = ratelimit.NewRateLimiter(reg, globalParams, bucketStore, logger)
 	}
 
+	if config.MaxBlobSize < 0 || config.MaxBlobSize > 64*1024*1024 {
+		return fmt.Errorf("configured max blob size is invalid %v", config.MaxBlobSize)
+	}
+
+	if int64(math.Log2(float64(config.MaxBlobSize))) == int64(math.Log2(float64(config.MaxBlobSize-1))) {
```
Comment (jianoaix) [inline disperser/cmd/apiserver/main.go:122]: It's faster to use bit operation, `config.MaxBlobSize > 0 && (config.MaxBlobSize&(config.MaxBlobSize-1)) == 0` should work

File: disperser/cmd/apiserver/main.go [hunk 2]
```diff
@@ -114,6 +115,14 @@ func RunDisperserServer(ctx *cli.Context) error {
 		ratelimiter = ratelimit.NewRateLimiter(reg, globalParams, bucketStore, logger)
 	}
 
+	if config.MaxBlobSize < 0 || config.MaxBlobSize > 64*1024*1024 {
```
Comment (jianoaix) [inline disperser/cmd/apiserver/main.go:118]: Should it fail if config.MaxBlobSize == 0?

File: disperser/cmd/apiserver/flags/flags.go [hunk 3]
```diff
@@ -94,6 +94,13 @@ var (
 		EnvVar:   common.PrefixEnvVar(envVarPrefix, "RATE_BUCKET_STORE_SIZE"),
 		Required: false,
 	}
+	MaxBlobSize = cli.IntFlag{
```
Comment (jianoaix) [inline disperser/cmd/apiserver/flags/flags.go:97]: It looks no issue if different disperser replicas are using different MaxBlobSize.
Should we add validation to the Node to check if the blob size is under a limit?

File: disperser/cmd/apiserver/main.go (hunk 2 above)
Comment (bxue-l2) [reply disperser/cmd/apiserver/main.go:118]: yeah, I will exclude 0

File: disperser/cmd/apiserver/flags/flags.go (hunk 3 above)
Comment (bxue-l2) [reply disperser/cmd/apiserver/flags/flags.go:97]: I don't think there is issue. But we want it, it has to be different deployment, which the current devops repo has yet offered that. 

I don't think we need to artificially constraint on blob size on the DA node size now, as far as the disperser isn't decentralized, but yes, once we got to the step. We need to have some rate limit or blob constraint. For now, I think we are good.

Empirically, there is no impact on validation speed with respect to the blob size.

File: disperser/cmd/apiserver/main.go (hunk 1 above)
Comment (bxue-l2) [reply disperser/cmd/apiserver/main.go:122]: this only runs once. I think it is more readable this way

Comment (bxue-l2) [reply disperser/cmd/apiserver/main.go:122]: but your solution works

Comment (jianoaix) [reply disperser/cmd/apiserver/main.go:122]: I think there are utility functions to check if a number is power of 2 (in encoding or common package). Reusing that has no impact on readabilty.

File: disperser/cmd/apiserver/main.go [hunk 4]
```diff
@@ -114,6 +115,14 @@ func RunDisperserServer(ctx *cli.Context) error {
 		ratelimiter = ratelimit.NewRateLimiter(reg, globalParams, bucketStore, logger)
 	}
 
+	if config.MaxBlobSize <= 0 || config.MaxBlobSize > 64*1024*1024 {
```
Comment (jianoaix) [inline disperser/cmd/apiserver/main.go:118]: So we want to support 64MiB?

File: disperser/cmd/apiserver/main.go (hunk 1 above)
Comment (bxue-l2) [reply disperser/cmd/apiserver/main.go:122]: that function is in "github.com/Layr-Labs/eigenda/encoding/fft". I don't want to add a new strange dependancy, what do you think

File: disperser/cmd/apiserver/main.go (hunk 4 above)
Comment (bxue-l2) [reply disperser/cmd/apiserver/main.go:118]: It is just a check, we might experiment with it

**Review by jianoaix — APPROVED** [review]

File: disperser/cmd/apiserver/main.go (hunk 1 above)
Comment (jianoaix) [reply disperser/cmd/apiserver/main.go]: ideally we can pull it out to `common/` and share across the codebase

**Pull request merged**